plugin/trino-accumulo/src/main/java/io/trino/plugin/accumulo/index/ColumnCardinalityCache.java

```diff
@@ -389,14 +388,14 @@ public Map<CacheKey, Long> loadAll(Iterable<? extends CacheKey> keys)
 
                 // Create a new map to hold our cardinalities for each range, returning a default of
                 // Zero for each non-existent Key
-                Map<CacheKey, Long> rangeValues = new HashMap<>();
+                ImmutableMap.Builder<CacheKey, Long> rangeValues = ImmutableMap.builder();
                 stream(keys).forEach(key -> rangeValues.put(key, 0L));
 
                 for (Entry<Key, Value> entry : scanner) {
                     rangeValues.put(rangeToKey.get(Range.exact(entry.getKey().getRow())), parseLong(entry.getValue().toString()));
                 }
 
-                return rangeValues;
+                return rangeValues.buildKeepingLast(); // TODO why duplicates here?
```

Comment:
cc @adamjshook

Reply:
I'll take a look.

Reply:
On line 392 `stream(keys).forEach(key -> rangeValues.put(key, 0L))` it creates an initial entry in the returned map for all `CacheKeys` set to zero. It then scans Accumulo overwriting the keys with the values in the database, leaving behind the default value when it is built for any missing keys. There will only be one key/value pair returned from Accumulo so `buildKeepingLast` is correct here. It could be rewritten to set any missing keys to `0` after the scan, though, if we want to use `buildOrThrow` instead.